# 🪵🍁🍂🍄🍗🦃🏈 Big Thanksgiving Dinner Analysis 🏈🦃🍗🍄🍂🍁🪵
**Name:** Johnathan Nguyen       <br>
**Class:** Python for Data Analytics

In [8]:
# Libraries
import numpy as np
import pandas as pd

In [9]:
# Synthetic Thanksgiving Data
np.random.seed()    # set seed to 2310
n = 50    # set n to 50

# creating the variables first (Age, Dinner, Dessert, Duration)
Age = (np.random.normal(37, 18, n).round(0).clip(5, 90))
Num_Dinner_Servings = np.round(4 - 0.0009 * (Age - 40)**2 + np.random.normal(0, 0.3, n)).clip(1, 5)
Num_Dessert_Servings = np.round(np.clip(2.5 - (Age / 50) + np.random.normal(0, 0.5, n), 0, 3))
Dinner_Duration_Hours =  np.round(1.5 + (Age / 40) + np.random.normal(0, 0.5, n), 1).clip(1, 5)

# creating boolean for whether or not they watched tv (older and younger people more likely to watch)
prob_parade = np.clip(    0.3 + 0.4 * np.exp(-((Age - 15)/20)**2) + 0.4 * np.exp(-((Age - 70)/20)**2),0, 1)
Watched_Parade = np.random.binomial(1, prob_parade)

# creating the data frame using the variables we just made
Thanksgiving = pd.DataFrame({
    'Ages' : Age,
    'Dinner_Servings' : Num_Dinner_Servings,
    'Dessert_Servings' : Num_Dessert_Servings,
    'Duration' : Dinner_Duration_Hours,
    'Macy_Parade' : Watched_Parade
})

Thanksgiving.head()


Ages  Dinner_Servings  Dessert_Servings  Duration  Macy_Parade
0  28.0              4.0               2.0       1.9            0
1  30.0              3.0               2.0       2.5            0
2  39.0              4.0               1.0       1.8            0
3  34.0              4.0               2.0       2.0            0
4  36.0              4.0               2.0       2.7            0

In [10]:
# Use the `describe` function to get the basic summary statistics
Thanksgiving.describe()

Ages  Dinner_Servings  Dessert_Servings   Duration  Macy_Parade
count  50.000000        50.000000          50.00000  50.000000    50.000000
mean   35.140000         3.620000           1.82000   2.366000     0.500000
std    17.227185         0.530306           0.66055   0.607642     0.505076
min     6.000000         2.000000           1.00000   1.100000     0.000000
25%    25.250000         3.000000           1.00000   1.925000     0.000000
50%    34.000000         4.000000           2.00000   2.250000     0.500000
75%    46.750000         4.000000           2.00000   2.800000     1.000000
max    80.000000         4.000000           3.00000   4.100000     1.000000

## Instructions:
Use this Thanksgiving dataset to explore patterns in the family dining habits.
You’ll need to _**interpret**_ the data using summary statistics, **not just report them**.

You will being using **only the describe function** to answer your questions

### Questions:

1. Looking at the minimum and maximum values of Age, what do they suggest about who attended this Thanksgiving dinner? <br>
**Answer:** The youngest person was 6 years old and the oldest was 80, suggesting there was a wide spread of ages that attended this dinner.<br>

2. Looking at the mean and standard deviation of Dinner_Duration, what do they show about how consistent the mealtime was across family members? <br>
**Answer:** It shows an average mean of 2.36 and a standard deviation of 0.61, showing that most people spent around the same time eating across family members.<br>

3. If someone stayed for 4.8 hours, would you consider them an outlier? _Use the mean and standard deviation to support your answer (Hint: $mean + 2*std$ OR $IQR=Q3−Q1$ and $Upper Bound=Q3+1.5×IQR$)._ <br>
**Answer:** Yes, they would be an outlier because using the equation given, the upper bound gives 3.73 hours, which means 4.8 is an outlier.<br>

4. Compare the mean and median of Dinner_Servings. What could a large difference between them indicate about family member behavior? _(Hint: 50% is half way in the data set)_ <br>
**Answer:** The mean (3.62) and median (4) are close, showing that the data is most likely symmetrical.<br>

5. Looking at the max value of Dessert_Servings, what type of guest do you imagine that might be and what could this tell us about dessert consumption overall?  <br>
**Answer:** It would most likely be a child or someone with a sweet tooth. This could show how spread out the data is due to the observations.<br>

6. Suppose the standard deviation of Dinner_Serving is small. What does that suggest about how much everyone ate?  <br>
**Answer:** Since the std of dinner servings is small, it suggests that everyone ate around the same. <br>

7. Which statistic (min, max, mean, std) best helps you find the “light eaters”? Explain how you would identify them and if the minimum makes sense.<br>
**Answer:** The min and mean would help find the light eaters because if they are closer to the min then they are lighter eaters compared to the mean.<br>

8. The range (max − min) of Dinner_Duration is quite large. What could that imply about how the dinner was organized?  <br>
**Answer:** The dinner could have been organized as a buffet-style, making some families stay longer compared to others. <br>

9. You suspect that older guests tend to eat fewer desserts. Could you use only the `describe` function to explore whether that idea might be true before doing a full analysis or would we have to include something else?  <br>
**Answer:** No, because you would need more than just the describe function, because you would need to see the correlation with Age vs Dessert_Servings<br>

10. What does the mean tell you about how common it was for people to watch the Macy’s Day Parade?  <br>
**Answer:** Since the mean is 0.5, it shows that half the group watched it, but it wasn't something everyone watched.<br>

11. what does the standard deviation suggest about the variation in watching behavior? Was parade-watching behavior fairly consistent across people, or was there a lot of difference between individuals?  <br>
**Answer:** The standard deviation of 0.51 shows that there was a wide split of people who did watch and people who didn't watch it at all. This shows mixed behavior in watching.<br>

12. In one sentence, summarize what kind of Thanksgiving this dataset represents. Focus on what the numbers imply rather than just stating the numbers.  <br>
**Answer:** The dataset represents a family Thanksgiving dinner with similar eating habits, meal durations, and an even split between people who watched the parade and people who didn't.<br>

13. Suggest some visuals that would be ideal for exploring this data and state their purpose. <br>
**Answer:** Histogram for Age and Dinner Duration to see if there is a trend with how long people stay at the table due to age, Scatter plot for age vs Dessert_Servings so we can see whether age has anything to do with the consumption,<br>
